hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -185,7 +185,9 @@ private void checkRecordKey(Configuration conf, ResolvedSchema schema) {
           && FlinkOptions.RECORD_KEY_FIELD.defaultValue().equals(recordKeys[0])
           && !fields.contains(recordKeys[0])) {
         throw new HoodieValidationException("Primary key definition is required, use either PRIMARY KEY syntax "
-            + "or option '" + FlinkOptions.RECORD_KEY_FIELD.key() + "' to specify.");
+            + "or option '" + FlinkOptions.RECORD_KEY_FIELD.key() + "' to specify. "
+            + "Otherwise the default primary key '"
```

Comment:
How about we strength it like this:

```java
Primary key definition is required, the default primary key field '{}' does not exist in the table schema, use either PRIMARY KEY syntax or option '{}' to speciy.
```

Reply:
I think your example is more reasonable and fluent, user can get the reason completely.

Reply:
Thanks, can you update it.

Reply:
Sure, I will do it today -.-